### PR TITLE
Update acknowledgment, glossary, NFR

### LINKED ISSUE #446: Acknowledgement section

### LINKED ISSUE #447: NFR: add in the tP constraints

### LINKED ISSUE #448: Glossary

### PR DESCRIPTION
**Pull request by zhonghan721**
Closes #446 
Closes #447 
Closes #448

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1824,19 +1830,30 @@ otherwise)
...
-- **Private Contact Detail**: A contact detail that is not meant to be shared with others
-- **CLI**: Command Line Interface
-- **Owner**: The customer who owns the home-based business and who uses the app
+| Term                           | Definition                                                                              |
+|--------------------------------|-----------------------------------------------------------------------------------------|
+| Alphanumeric                   | Consisting of only letters and numbers                                                  |
+| Command Line Interface (CLI)   | A text-based user interface used to run programs                                        |
+| Graphical User Interface (GUI) | A visual interface where you can interact with the program through graphical components |
+| JSON                           | Short for JavaScript Object Notation, a lightweight format for storing your data        |
+| Owner                          | The customer who owns the home-based business and who uses the app                      |
```
Comment (jianyangg) [inline docs/DeveloperGuide.md]: Should the definition be "The individual who owns the home-based business and who uses the HomeBoss app."? I think I made this mistake in the previous version of the Glossary.

**Review by jianyangg — COMMENTED** [review]
Just one comment, otherwise LGTM.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (zhonghan721) [reply docs/DeveloperGuide.md]: Updated!

**Review by juliusgambe — APPROVED** [review]
LGTM

**Pull request merged**